core/src/Threads/Kokkos_Threads_Instance.cpp

```diff
@@ -278,12 +278,11 @@ void ThreadsInternal::execute_sleep(ThreadsInternal &exec, const void *) {
   const int rank_rev = exec.m_pool_size - (exec.m_pool_rank + 1);
 
   for (int i = 0; i < n; ++i) {
-    Impl::spinwait_while_equal<int>(
-        exec.m_pool_base[rank_rev + (1 << i)]->m_pool_state,
-        ThreadsInternal::Active);
+    Impl::spinwait_while_equal<State>(
+        exec.m_pool_base[rank_rev + (1 << i)]->m_pool_state, State::Active);
```

Comment:
```suggestion
    Impl::spinwait_while_equal(m_pool_base[rev_rank + (1 << i)]->m_pool_state,
                               State::Active);
```
Same comment everywhere.  I don't think there is any point on specifying the template parameters.